mesa/space.py

```diff
@@ -414,24 +435,36 @@ def move_agent(self, agent: Agent, pos: Coordinate) -> None:
             pos: Tuple of new position to move the agent to.
         """
         pos = self.torus_adj(pos)
-        self.remove_agent(agent)
-        self.place_agent(agent, pos)
+        if self.track_empties:
+            self._empties.discard(pos)
+            self._empties.add(agent.pos)
+        x, y = agent.pos
```

Comment:
This is reinventing the wheel. Why replace `remove_agent` and `place_agent`?

Reply:
because it has 2 advantages:

- when `track_empties` is False we don't do anything on `num_empties` because it remains unchanged
- `agent.pos` is not reset twice
- we spare two functions calls

but let me know if you think those are negligible

Reply:
The priority is readability > performance, unless the performance benefit is very significant. There is also the burden of having to explain why it is done this way instead of the other, for future readers of the code.